### ISSUE TITLE
Calling async functions causes lots of Monti APM endevent forced complete

### ISSUE THREAD

**Issue by lynchem** [monti-apm/monti-apm-agent#3]
Note: I believe Kadira has some issue handling async calls. There was some related activity on the meteor apm repo a few months back. 
https://github.com/meteor/meteor-apm-agent/pull/5
Here glasser sheds some light on it: https://github.com/meteor/meteor/issues/9730

Anyway, we have an async function which calls the google API

export const getGeoCoords = async function(addressDoc) ... 

And then we call the function inside the collection hook that looks something like this: 

```
OurCollection.after.update(function(userId, doc, fieldNames, modifier, options) {
        getGeoCoords(filteredAddress)
                .then(result => {
                    OurCollection.update(doc._id, mod, {bypassCollection2: true});
                })
});
```

Here's what it prints in the console
```
[
    {
        "type": "start",
        "at": 1529664466943,
        "data": {
            "userId": "theUsersId",
            "params": "[{},true]"
        }
    },
    {
        "type": "wait",
        "at": 1529664466943,
        "data": {}
    },
    {
        "type": "waitend",
        "at": 1529664466943,
        "data": {
            "waitOn": []
        }
    },
    {
        "type": "db",
        "at": 1529664466944,
        "data": {
            "coll": "users",
            "func": "find",
            "selector": "\"theUsersId\""
        }
    },
    {
        "type": "dbend",
        "at": 1529664466944,
        "data": {}
    },
    {
        "type": "db",
        "at": 1529664466944,
        "data": {
            "coll": "users",
            "selector": "{\"_id\":\"theUsersId\"}",
            "func": "fetch",
            "cursor": true,
            "limit": 1
        }
    },
    {
        "type": "dbend",
        "at": 1529664466946,
        "data": {
            "docsFetched": 1,
            "docSize": 3053
        }
    },
    {
        "type": "db",
        "at": 1529664466947,
        "data": {
            "coll": "users",
            "func": "find",
            "selector": "{\"_id\":\"theUsersId\"}"
        }
    },
    {
        "type": "dbend",
        "at": 1529664466947,
        "data": {}
    },
    {
        "type": "db",
        "at": 1529664466947,
        "data": {
            "coll": "users",
            "selector": "{\"_id\":\"theUsersId\"}",
            "func": "fetch",
            "cursor": true,
            "limit": 1
        }
    },
    {
        "type": "dbend",
        "at": 1529664467000,
        "data": {
            "docsFetched": 1,
            "docSize": 3053
        }
    },
    {
        "type": "db",
        "at": 1529664467002,
        "data": {
            "coll": "places",
            "func": "insert"
        }
    },
    {
        "type": "dbend",
        "at": 1529664467010,
        "data": {}
    },
    {
        "type": "db",
        "at": 1529664467011,
        "data": {
            "coll": "users",
            "func": "find",
            "selector": "\"theUsersId\""
        }
    },
    {
        "type": "dbend",
        "at": 1529664467011,
        "data": {}
    },
    {
        "type": "db",
        "at": 1529664467011,
        "data": {
            "coll": "users",
            "selector": "{\"_id\":\"theUsersId\"}",
            "func": "fetch",
            "cursor": true,
            "limit": 1
        }
    },
    {
        "type": "dbend",
        "at": 1529664467013,
        "data": {
            "docsFetched": 1,
            "docSize": 3053
        }
    },
    {
        "type": "db",
        "at": 1529664467013,
        "data": {
            "coll": "users",
            "func": "find",
            "selector": "{\"_id\":\"theUsersId\"}"
        }
    },
    {
        "type": "dbend",
        "at": 1529664467014,
        "data": {}
    },
    {
        "type": "db",
        "at": 1529664467014,
        "data": {
            "coll": "users",
            "selector": "{\"_id\":\"theUsersId\"}",
            "func": "fetch",
            "cursor": true,
            "limit": 1
        }
    },
    {
        "type": "dbend",
        "at": 1529664467016,
        "data": {
            "docsFetched": 1,
            "docSize": 3053
        }
    },
    {
        "type": "db",
        "at": 1529664467018,
        "data": {
            "coll": "logs",
            "func": "insert"
        }
    },
    {
        "type": "dbend",
        "at": 1529664467108,
        "data": {}
    },
    {
        "type": "async",
        "at": 1529664467108
    },
    {
        "type": "asyncend",
        "at": 1529664467108
    },
    {
        "type": "async",
        "at": 1529664467313
    },
    {
        "type": "asyncend",
        "at": 1529664467410
    }
]
```

**Comment by lc3t35** [comment]
I've submit this PR to limit the warning on console.
I've noticed you have a configuration problem on travis ci for meteor 1.6.1 and 1.7rc2, test pass of 1.4 and 1.5 : Error: Cannot find module '@babel/runtime/helpers/builtin/interopRequireDefault', look at https://github.com/meteor/meteor/issues/10128

**Comment by zodern** [comment]
I have started work on this.

**Comment by zodern** [comment]
I published version 2.33.0 today to remove the warnings. Next month I will improve how async calls are handled.
